tests/build-post-list.test.js

```diff
@@ -0,0 +1,253 @@
+const fs = require('fs-extra');
+const { resolve, join } = require('path');
+const { TEST_CONTENT } = require('../tests/fixtures/buildPostListData');
+const { buildPostList, slugifyToC } = require('../scripts/build-post-list');
+
+describe('buildPostList', () => {
+  let tempDir;
+  let writeFilePath;
+  let postDirectories;
+
+  beforeEach(async () => {
+    tempDir = resolve(
+      __dirname,
+      `test-config-${Date.now()}-${Math.random().toString(36).slice(2)}`,
+    );
+    writeFilePath = resolve(tempDir, 'posts.json');
+    postDirectories = [
+      [join(tempDir, 'blog'), '/blog'],
+      [join(tempDir, 'docs'), '/docs'],
+      [join(tempDir, 'about'), '/about'],
+    ];
+
+    await fs.ensureDir(join(tempDir, TEST_CONTENT.blog.dir));
+    await fs.writeFile(
+      join(tempDir, TEST_CONTENT.blog.dir, TEST_CONTENT.blog.file),
+      TEST_CONTENT.blog.content,
+    );
+
+    await fs.ensureDir(join(tempDir, TEST_CONTENT.docs.dir));
+    await fs.writeFile(
+      join(tempDir, TEST_CONTENT.docs.dir, TEST_CONTENT.docs.file),
+      TEST_CONTENT.docs.content,
+    );
+
+    await fs.ensureDir(
+      join(tempDir, TEST_CONTENT.docs.dir, TEST_CONTENT.docs.subDir),
+    );
+
+    await fs.ensureDir(join(tempDir, TEST_CONTENT.about.dir));
+    await fs.writeFile(
+      join(tempDir, TEST_CONTENT.about.dir, TEST_CONTENT.about.file),
+      TEST_CONTENT.about.content,
+    );
```

Comment:
Move this implementation to a separate function, beforeEach setup is quite lengthy and could be moved to a separate helper function for better maintainability. Use this:

```
+async function setupTestDirectories(tempDir) {
  const dirs = ['blog', 'docs', 'about'];
  for (const dir of dirs) {
    await fs.ensureDir(join(tempDir, TEST_CONTENT[dir].dir));
    await fs.writeFile(
      join(tempDir, TEST_CONTENT[dir].dir, TEST_CONTENT[dir].file),
      TEST_CONTENT[dir].content
    );
  }
  await fs.ensureDir(join(tempDir, TEST_CONTENT.docs.dir, TEST_CONTENT.docs.subDir));
}
```

Reply:
Done